pyomo/dae/flatten.py

```diff
@@ -74,7 +74,7 @@ def generate_time_only_slices(obj, time):
     )
     # For each combination of regular indices, we can generate a single
     # slice over the time index
-    time_sliced = {time_idx: time.first()}
+    time_sliced = {time_idx: slice(None)}
```

Comment:
Sliced indices should map to `slice(None)`, no? Seems to work regardless, which slightly confuses me, but I guess the value is unimportant here as long as you know what index to slice over.

Reply:
This actually only needs to be a list.  If you dive into the bowls of the slice logic, the only thing that is actually necessary is it's length, however, I think I would recommend changing this to:
```python
time_sliced = [time_idx]
```